Test/separate-verification/assumptions.dfy.expect

```diff
@@ -0,0 +1,21 @@
+TestAuditor.dfy(87,4): Warning: note, this forall statement has no body
+TestAuditor.dfy(94,4): Warning: note, this loop has no body (loop frame: i)
+TestAuditor.dfy(85,10): Warning: /!\ No terms found to trigger on.
+TestAuditor.dfy(87,4): Warning: /!\ No terms found to trigger on.
+
+Dafny program verifier did not attempt verification
+assumptions.dfy(6,8): Error: the included file TestAuditor.dfy contains error(s)
+TestAuditor.dfy(18,6): Error: Ghost declaration has no body and has an ensures clause.
+TestAuditor.dfy(23,15): Error: Ghost declaration has no body and has an ensures clause.
+TestAuditor.dfy(41,17): Error: Compiled declaration has no body and has an ensures clause.
+TestAuditor.dfy(41,17): Error: Declaration with [{:extern}] has a ensures clause.
```

Comment:
I'm on the fence about this one. Do you think the standard for published libraries should be that they include only `bare` externs, and use something like `Need` to concretely check? Or should we instead allow contracts on externs but require that they're (automatically) tested, with `--test-assumptions Externs`?

Reply:
I was aiming to default to the safer option for now. It'll let us publish `FileIO` from `dafny-lang/libraries` for example, but not `MutableMap`. I'd like to allow them too, but I'm not yet sure how to enforce at the time of building with `-t:lib` that you ran with `--test-assumptions Externs` "at some point".

Plus it's probably helpful to distinguish in some way as a consumer if you're consuming a "fully verified" library or not. Perhaps you can run the auditor in such a way that libraries are in scope too?

Reply:
Yeah, I think being stricter at first and then possibly making thing more lenient in the future is a good strategy.

Reply:
Yeah, Haskell has a notion of "safe" modules, which have more restricted functionality (IIRC, this include no FFI, no unsafe things, etc.) and can only import other safe modules. We could imagine having different categories of libraries, and the first one we implement could be the strictest.

Reply:
I actually anticipate needing an exact match on the Dafny version is going to be the biggest initial pain.